src/solver/expressions/include/antares/solver/expressions/visitors/LinearStatus.h

```diff
@@ -20,121 +20,112 @@
 */
 #pragma once
 
+#include <ostream>
+
 namespace Antares::Solver::Visitors
 {
 
-class LinearStatus
+enum class LinearStatus : char
 {
-public:
-    enum Status
-    {
-        CONSTANT,
-        LINEAR,
-        NON_LINEAR
-    };
-
-    constexpr LinearStatus() = default;
-
-    constexpr LinearStatus(const Status& status):
-        status_(status)
-    {
-    }
-
-    constexpr LinearStatus(const LinearStatus& other) = default;
-
-    constexpr LinearStatus operator*(const LinearStatus& other)
-    {
-        switch (other)
-        {
-        case LinearStatus::NON_LINEAR:
-            return LinearStatus::NON_LINEAR;
-        case LinearStatus::CONSTANT:
-            return *this;
-        case LinearStatus::LINEAR:
-            if (status_ == LinearStatus::CONSTANT)
-            {
-                return other;
-            }
-            else
-            {
-                return LinearStatus::NON_LINEAR;
-            }
-        default:
-            return LinearStatus::NON_LINEAR;
-        }
-    }
-
-    constexpr LinearStatus operator/(const LinearStatus& other)
-    {
-        switch (other)
-        {
-        case LinearStatus::NON_LINEAR:
-        case LinearStatus::LINEAR:
-            return LinearStatus::NON_LINEAR;
-        case LinearStatus::CONSTANT:
-            return *this;
-        default:
-            return LinearStatus::NON_LINEAR;
-        }
-    }
-
-    constexpr LinearStatus operator+(const LinearStatus& other)
-    {
-        switch (other)
-        {
-        case LinearStatus::NON_LINEAR:
-            return LinearStatus::NON_LINEAR;
-        case LinearStatus::CONSTANT:
-            return *this;
-        case LinearStatus::LINEAR:
-            if (other == LinearStatus::CONSTANT || other == LinearStatus::LINEAR)
-            {
-                return other;
-            }
-            else
-            {
-                return LinearStatus::NON_LINEAR;
-            }
-        default:
-            return LinearStatus::NON_LINEAR;
-        }
-    }
-
-    constexpr LinearStatus operator-(const LinearStatus& other)
-    {
-        return operator+(other);
-    }
-
-    // Conversions
-    constexpr explicit operator bool() const = delete;
+    CONSTANT = 0,
+    LINEAR = 1,
+    NON_LINEAR = 2
+};
 
-    constexpr operator Status() const
-    {
-        return status_;
-    }
+constexpr char pair(LinearStatus a, LinearStatus b)
+{
+    return static_cast<char>(a) << 4 | static_cast<char>(b);
+}
 
-    // Comparisons
-    constexpr bool operator==(LinearStatus a) const
+constexpr LinearStatus operator*(LinearStatus a, LinearStatus b)
```

Comment:
As **LinearStatus::NON_LINEAR** is the default case, we could ignore cases where we return **LinearStatus::NON_LINEAR**.
If you think this leads to a lack of clarity, ignore this comment.

Reply:
I think it's best to keep all cases, for clarity.